### ISSUE TITLE
Implement remaining healpy compatibility functions

### ISSUE THREAD

**Issue by astrofrog** [cdeil/healpix#23]
We can still add compatibility functions (just thin wrappers) for the following:

* [ ] ``healpy.boundaries`` (using ``healpix_boundaries_lonlat``)
* [ ] ``healpy.nest2ring`` (using ``nested_to_ring``)
* [ ] ``healpy.ring2nest`` (using ``ring_to_nested``)
* [ ] ``healpy.reorder`` (using ``nested_to_ring`` or ``ring_to_nested``)
* [ ] ``get_interp_val`` (using ``interpolate_bilinear_lonlat``)
* [ ] ``query_disc`` (using ``cone_search_lonlat``)

**Comment by astrofrog** [comment]
I don't think we want to implement ``query_disk`` since we use a different algorithm that seems to give more reliable results so no point trying to reproduce the healpy algorithm.

**Comment by cdeil** [comment]
Here's what we have now:
http://astropy-healpix.readthedocs.io/en/latest/api.html#module-astropy_healpix.healpy

As mentioned, `reorder` is still missing:
http://healpy.readthedocs.io/en/latest/generated/healpy.pixelfunc.reorder.html

@astrofrog - Do you think we should add it now, or wait for users to request it?
@lpsinger - Do you use it? Or are there others that are missing that we should add before cutting a new release?

**Comment by astrofrog** [comment]
@cdeil - reorder should be easy enough to implement by making a cython function that calls ``healpixl_nested_to_xy`` then ``healpixl_xy_to_ring`` (or vice versa). If no one has time to work on it now, then no harm in waiting until people ask for it.

**Comment by lpsinger** [comment]
> Do you use it?

Yes I do.

> Or are there others that are missing that we should add before cutting a new release?

Not sure at the moment; it would take me an hour or two to update astropy-healpix branches of my code and test them.
